src/test/java/com/commercetools/sync/cartdiscounts/utils/CartDiscountSyncUtilsTest.java

```diff
@@ -0,0 +1,121 @@
+package com.commercetools.sync.cartdiscounts.utils;
+
+import com.commercetools.sync.cartdiscounts.CartDiscountSyncOptions;
+import com.commercetools.sync.cartdiscounts.CartDiscountSyncOptionsBuilder;
+import io.sphere.sdk.cartdiscounts.CartDiscount;
+import io.sphere.sdk.cartdiscounts.CartDiscountDraft;
+import io.sphere.sdk.cartdiscounts.CartDiscountDraftBuilder;
+import io.sphere.sdk.cartdiscounts.CartDiscountTarget;
+import io.sphere.sdk.cartdiscounts.CartDiscountValue;
+import io.sphere.sdk.cartdiscounts.CartPredicate;
+import io.sphere.sdk.cartdiscounts.LineItemsTarget;
+import io.sphere.sdk.cartdiscounts.ShippingCostTarget;
+import io.sphere.sdk.cartdiscounts.StackingMode;
+import io.sphere.sdk.cartdiscounts.commands.updateactions.ChangeCartPredicate;
+import io.sphere.sdk.cartdiscounts.commands.updateactions.ChangeIsActive;
+import io.sphere.sdk.cartdiscounts.commands.updateactions.ChangeName;
+import io.sphere.sdk.cartdiscounts.commands.updateactions.ChangeRequiresDiscountCode;
+import io.sphere.sdk.cartdiscounts.commands.updateactions.ChangeSortOrder;
+import io.sphere.sdk.cartdiscounts.commands.updateactions.ChangeStackingMode;
+import io.sphere.sdk.cartdiscounts.commands.updateactions.ChangeTarget;
+import io.sphere.sdk.cartdiscounts.commands.updateactions.ChangeValue;
+import io.sphere.sdk.cartdiscounts.commands.updateactions.SetDescription;
+import io.sphere.sdk.cartdiscounts.commands.updateactions.SetValidFromAndUntil;
+import io.sphere.sdk.client.SphereClient;
+import io.sphere.sdk.commands.UpdateAction;
+import io.sphere.sdk.models.LocalizedString;
+import io.sphere.sdk.utils.MoneyImpl;
+import org.junit.Test;
+
+import java.time.ZonedDateTime;
+import java.util.List;
+import java.util.Locale;
+
+import static io.sphere.sdk.json.SphereJsonUtils.readObjectFromResource;
+import static io.sphere.sdk.models.DefaultCurrencyUnits.EUR;
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.mockito.Mockito.mock;
+
+public class CartDiscountSyncUtilsTest {
+
+    private static final SphereClient CTP_CLIENT = mock(SphereClient.class);
+    private static final CartDiscount CART_DISCOUNT_WITH_SHIPPING_TARGET =
+        readObjectFromResource("cart-discount-with-shipping-target.json", CartDiscount.class);
+
+    @Test
+    public void buildActions_FromDraftsWithDifferentNameValues_ShouldBuildUpdateActions() {
+        final LocalizedString newName =
+            LocalizedString.of(Locale.GERMAN, "Neu Name", Locale.ENGLISH, "new name");
+
+        final CartDiscountDraft newCartDiscount = CartDiscountDraftBuilder
```

Comment:
Is it possible to create a draft from `CART_DISCOUNT_WITH_SHIPPING_TARGET` and change only what you need? It's quite hard to review what's the difference between this newCartDiscount and `CART_DISCOUNT_WITH_SHIPPING_TARGET` and why you have only one action `ChangeName` returned.

Reply:
I could not deep copy from the CART_DISCOUNT_WITH_SHIPPING_TARGET but I hope it's more readable now.
 
Commit: 5b761ff

Reply:
I definitley agree with @lojzatran comment here. It's really hard throughout the rest of the unit tests.